### PR TITLE
Added regex matching for wood cutting & Add WoodcuttingPluginTest (2 tests)

### LINKED ISSUE #10212: Arctic Pine Logs & Woodcutting plugin
Arctic pine logs are not tracked by the woodcutting plugin.

Courtesy of @Alexsuperfly the tracked game message pattern of

`if (event.getMessage().startsWith("You get some") && (event.getMessage().endsWith("logs.") || event.getMessage().endsWith("mushrooms.")))`

doesn't capture the game message

`You get an arctic pine log.`

Would appreciate if this could get added.

### PR DESCRIPTION
**Pull request by JZomerlei**
Closes #10212 

I was trying to get regex to work but my small brain couldn't seem to get 
message to regex for matches on "^You get (some|an)" && matches on logs?.$

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public void onChatMessage(ChatMessage event)
 	{
 		if (event.getType() == ChatMessageType.SPAM || event.getType() == ChatMessageType.GAMEMESSAGE)
 		{
-			if (event.getMessage().startsWith("You get some") && (event.getMessage().endsWith("logs.") || event.getMessage().endsWith("mushrooms.")))
+			if (event.getMessage().matches("You get (?:some|an)[\\w ]+(?:logs?|mushrooms)\\."))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java]: precompile the pattern and save it as constant at top of the file

Comment (JZomerlei) [reply runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java]: Good idea.

File: runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java [hunk 2]
```diff
@@ -87,6 +88,8 @@
 	@Getter
 	private final Set<GameObject> treeObjects = new HashSet<>();
 
+	private static final Pattern WOOD_CUT_PATTERN = Pattern.compile("You get (?:some|an)[\\w ]+(?:logs?|mushrooms)\\.");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java]: Move this to top of the class, before all non-static stuff

**Comment by Adam-** [comment]
It would be good to additionally have a test for the various messages.

**Comment by JZomerlei** [comment]
> It would be good to additionally have a test for the various messages.

Added 2 tests.

File: runelite-client/src/test/java/net/runelite/client/plugins/woodcutting/WoodcuttingPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import net.runelite.client.plugins.xptracker.XpTrackerService;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/woodcutting/WoodcuttingPluginTest.java]: The import block of this file is a bit messed up. Remove the `*` imports and empty lines in it.

**Comment by JZomerlei** [comment]
Yes or no to keep the duration of session test? Could it ever fail if some how the CPU is just really really faster in comparing the instant a few lines apart? I've ran the test ~150 times local computer and it never failed but I don't think it's a particularly good test to include. 

I was simply going a tiny bit overboard.

**Comment by deathbeam** [comment]
If you want to test the session timeout, then simply add sleep to the test

**Comment by JZomerlei** [comment]
> If you want to test the session timeout, then simply add sleep to the test

Adam mentioned to not use Thread.sleep(), but maybe 5 wouldn't hurt. Originally it was 60001 to test for when the session time out was set to 1, but now it's set to 0.

I'll add 5ms.

**Comment by deathbeam** [comment]
Why not? I used it for testing configured timeouts for IdleNotifierPlugin, Thread.sleep is perfectly fine in tests.

**Comment by JZomerlei** [comment]
I think it's more of I originally showed that Thread.sleep(60001) would work, 60 seconds for a test is a bit steep. Is it not?

File: runelite-client/src/test/java/net/runelite/client/plugins/woodcutting/WoodcuttingPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.xptracker.XpTrackerService;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
```
[the remark this answers is not in the document]
Comment (JZomerlei) [reply runelite-client/src/test/java/net/runelite/client/plugins/woodcutting/WoodcuttingPluginTest.java]: The code style says they’re blocked with a blank line between them.

**Pull request merged**